runtime/src/commonMain/kotlin/pbandk/FieldDescriptor.kt

```diff
@@ -125,4 +132,52 @@ class FieldDescriptor<M : Message, T>(
             override fun isDefaultValue(value: Any?) = (value as? kotlin.collections.Map<*, *>)?.isEmpty() == true
         }
     }
+
+    @Suppress("UNCHECKED_CAST")
+    fun <M : Message, T> parseUnknownField(
+        unknownField: UnknownField.Value,
+        fieldDescriptor: FieldDescriptor<M, T>
+    ): T {
+        return when (fieldDescriptor.type) {
+            is Type.Primitive<*> -> parseUnknownFieldForPrimitiveOrComposite(fieldDescriptor.type, unknownField) as T
+            is Type.Message<*> -> parseUnknownFieldForTypeMessage(fieldDescriptor.type, unknownField) as T
+            else -> throw UnsupportedOperationException("fieldDescriptor.type: '${fieldDescriptor.type}' is not supported")
```

Comment:
I think we need to support at least `Type.Enum` too. But probably `Type.Repeated` and `Type.Map` also since I'd guess those can be used as extension fields.